detection_rules/mixins.py

```diff
@@ -103,9 +127,9 @@ def jsonschema(cls):
         return jsonschema
 
     @classmethod
-    def from_dict(cls: Type[ClassT], obj: dict) -> ClassT:
+    def from_dict(cls: Type[ClassT], obj: dict, unknown: Optional[bool] = None) -> ClassT:
```

Comment:
nit: 
```suggestion
    def from_dict(cls: Type[ClassT], obj: dict, skip_unknown: Optional[bool] = None) -> ClassT:
```

Reply:
changing the type from bool but rejecting this suggestion. The values are: include, exclude, raise